lua/treesitter-context.lua

```diff
@@ -392,31 +392,27 @@ end
 
 --- @generic F: function
 --- @param fn F
+--- @param ms? number
 --- @return F
-local function throttle_fn(fn)
-  local recalc_after_cooldown = false
-  local cooling_down = false
-  local function wrapped()
-    if cooling_down then
-      recalc_after_cooldown = true
+local function debounce(fn, ms)
+  ms = ms or 20
+  fn = vim.schedule_wrap(fn)
+  local timer = vim.loop.new_timer()
+  local waiting = 0
+  return function()
+    if waiting == 0 then
+      fn() -- first call, execute immediately
     else
-      local start = vim.loop.hrtime()
-      fn()
-      local elapsed_ms = math.floor((vim.loop.hrtime() - start) / 1e6)
-      -- If this took < 2ms, we don't need a cooldown period. This prevents the context floats from flickering
-      if elapsed_ms > 2 then
-        cooling_down = true
-        vim.defer_fn(function()
-          cooling_down = false
-          if recalc_after_cooldown then
-            recalc_after_cooldown = false
-            wrapped()
-          end
-        end, 20)
-      end
+      waiting = waiting + 1
```

Comment:
This never gets executed, so the debounce logic never engages

Reply:
Good call! :)